type-generator/src/main/scala/mesosphere/raml/backend/treehugger/FieldVisitor.scala

```diff
@@ -10,9 +10,15 @@ import scala.annotation.tailrec
 
 object FieldVisitor {
 
-  def visit(params: Seq[FieldT]): Seq[ValDef] = {
+  def visit(params: Seq[FieldT], flags: Seq[Long] = Seq()): Seq[ValDef] = {
```

Comment:
What are flags?

Reply:
Flags are modifier for the Field-Definitions, i.e. FINAL, PARAM, OVERRIDE, etc. 

In this case, we need to specify the fields with OVERRIDE when we're extending a trait.

Reply:
And the flags are longs? Hm, could you add docs?